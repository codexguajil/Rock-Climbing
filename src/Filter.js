import React, { Component } from 'react';

export default class Filter extends Component {
  constructor() {
    super();
    this.state = {
      setMax: 50,
      types: [],
      areas: [],
      areasPick: [],
      typesPick: []
    }
  }

  changeType = event => {
    console.log(this.props.climb)
    let type = Object.keys(this.props.climb).reduce((acc, area) => {
      if(this.props.climb[area].milesFromDenver <= event.target.value){
        this.props.climb[area].typesOfClimbing.forEach((type) => {
          if(!acc.includes(type)){
            acc.push(type)
          }
        })
      }
      return acc;
    }, [])
    this.setState({
      setMax: event.target.value,
      types: type
    })
    this.changeArea()
  }

  changeArea = event => {
    let types = []
    document.querySelectorAll('input[name="types"]:checked').forEach((type) => {
      types.push(type.value)
    })
    let areas = Object.keys(this.props.climb).reduce((acc, area) => {
      if(this.props.climb[area].milesFromDenver <= this.state.setMax){
        types.forEach((type) => {
          if(this.props.climb[area].typesOfClimbing.includes(type) && !acc.includes(area)){
            acc.push(area)
          }
        })
      }
      return acc;
    }, [])
    this.setState({
      areas: areas,
      typesPick: types
    })
  }

  UpdateAreasPick = () => {
    let areas = []
    document.querySelectorAll('input[name="areas"]:checked').forEach((type) => {
      areas.push(type.value)
    })
    this.setState({
      areasPick: areas
    })
  }

  submitFilter = event => {
    event.preventDefault()
    console.log('submit button')
    this.props.updateFromFilter(this.state.areasPick, this.state.typesPick)
  }


  render() {
    return (
      <form onSubmit={this.submitFilter} className="filter-section"> 
        <h2>Pick Your Next Plan</h2>
        <h3>Distance From Denver</h3>
        <div className="filter-range-section">
          <p>{this.state.setMax} miles away</p>
          <input className="range" type="range" onChange={this.changeType} min="1" max="400" ></input>
        </div>
        <h3>Type</h3>
        <div className="filter-type-section">
          {
            this.state.types.map((type) => {
              return (
                <div>
                  <input onChange={this.changeArea} name="types" type="checkbox" value={type}/> {type}
                </div>
              )
            })
          }
        </div>
        <h3>Areas</h3>
        <div className="filter-area-section">
          {
            this.state.areas.map((area) => {
              return (
                <div>
                  <input onChange={this.UpdateAreasPick} name="areas" type="checkbox" value={area}/> {area}
                </div>
              )
            })
          }
        </div>
        <input class="filter-submit-button" type="submit" value="Climb"></input>
      </form>
    )
  }
}