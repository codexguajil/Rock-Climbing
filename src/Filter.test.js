import React from 'react';
import Filter from './Filter';
import { shallow } from 'enzyme';

  const mockColoradoClimbingAreas = {
    "Canon City": {
        "milesFromDenver": 119,
        "id": 34,
        "typesOfClimbing": ["trad", "sport", "toprope", "boulder"],
        "classicRoutes": [
            {
                "name": "Enterprise",
                "id": 35,
                "grade": "5.09+",
                "type": ["sport"],
                "link": "https://www.mountainproject.com/route/105824687/enterprise"
            },
            {
                "name": "Trick-or-Treat",
                "id": 42,
                "grade": "5.11a/b",
                "type": ["toprope"],
                "link": "https://www.mountainproject.com/route/105759012/trick-or-treat"
            }
        ]
    },
    "Boulder Canyon": {
        "milesFromDenver": 31.5,
            "id": 12,
            "typesOfClimbing": ["trad", "sport", "toprope", "boulder", "aid"],
            "classicRoutes": [
                {
                    "name": "North Face Center",
                    "id": 13,
                    "grade": "5.07+",
                    "type": ["trad"],
                    "link": "https://www.mountainproject.com/route/105748093/north-face-center"
                },
                {
                    "name": "Empor",
                    "id": 14,
                    "grade": "5.07+",
                    "type": ["trad"],
                    "link": "https://www.mountainproject.com/route/105748142/empor"
                }
            ]
        }
    }


  const mockFunction = jest.fn();






describe('Filter', () => {

  let wrapper;


  beforeEach(() => {
    wrapper = shallow(
      <Filter climb={mockColoradoClimbingAreas}
                    updateFromFilter={mockFunction}/>
      )
  })

  it('should test the states', () => {
    expect(wrapper.state()).toEqual({setMax: 50, types: [], areas: [], areasPick: [], typesPick: []
    })
  })

  it('should change type', () => {
    expect(wrapper.state('setMax')).toEqual(50);
    expect(wrapper.state('types')).toEqual([]);
    wrapper.find('.range').simulate('change', {target: {value: 115}});
    expect(wrapper.state('setMax')).toEqual(115);
    expect(wrapper.state('types')).toEqual(["trad", "sport", "toprope", "boulder", "aid"]);
  })

  it.only('should setState for typesPick and areas', () => {
    expect(wrapper.state('typesPick')).toEqual([]);
    expect(wrapper.state('areas')).toEqual([]);
    wrapper.find('.range').simulate('change', {target: {value: 120}});
    wrapper.find('.types-checkbox').at(0).simulate('change', {target: {checked: true}})
    wrapper.instance().changeArea();
    // expect(wrapper.state('typesPick')).toEqual(["trad", "sport", "toprope", "boulder", "aid"]);
    expect(wrapper.state('areas')).toEqual(['Canon City','Boulder Canyon']);
  })

  it('should Update Area', () => {
    // areasPick
    expect.state('areasPick').toEqual([]);
    wrapper.find('.range').simulate('change', {target: {value: 120}});
    expect.state('areasPick').toEqual(['Canon City', 'Boulder Canyon'])
  })


  it('render with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})