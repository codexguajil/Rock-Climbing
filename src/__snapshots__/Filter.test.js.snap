// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter render with snapshot 1`] = `
<form
  className="filter-section"
  onSubmit={[Function]}
>
  <h2>
    Pick Your Next Plan
  </h2>
  <h3>
    Distance From Denver
  </h3>
  <div
    className="filter-range-section"
  >
    <p>
      50
       miles away
    </p>
    <img
      alt="hiker"
      className="hiker-image"
      src="hiker.png"
    />
    <input
      className="range"
      max="400"
      min="1"
      onChange={[Function]}
      type="range"
    />
  </div>
  <h3>
    Type
  </h3>
  <div
    className="filter-type-section"
  />
  <h3>
    Areas
  </h3>
  <div
    className="filter-area-section"
  />
  <input
    class="filter-submit-button"
    type="submit"
    value="Climb"
  />
</form>
`;
